install/0000_30_machine-api-operator_07_machinehealthcheck.crd.yaml

```
@@ -73,7 +73,7 @@ spec:
                   unsigned duration string of decimal numbers each with optional fraction
                   and a unit suffix, eg "300ms", "1.5h" or "2h45m". Valid time units
                   are "ns", "us" (or "µs"), "ms", "s", "m", "h".
-                pattern: ^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h)*)+$
+                pattern: ^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h))+$
```

is the suffix optional?

---

Suffix is not optional, not including a suffix causes the duration parse to fail, which is a bug we were suffering in previous releases as well

See https://play.golang.org/p/ZnSApQwLCgK

---

just wondering if we should tighten the regex by adding a qualifier?

```suggestion
                pattern: ^([0-9]+(\.[0-9]+)?(ns|us|µs|ms|s|m|h){1})+$
```

---

I don't think adding either of these is correct:
- `(ns|us|µs|ms|s|m|h)` means that we must match one of these strings exactly, so for example `mss` wouldn't be a match, adding `{1}` to this doesn't make it any stronger (https://regexr.com/59jkk)
- `3h5m` is actually a valid duration that can be parsed, so swapping the final `+` for a `{1}` would restrict the allowed durations to a single unit'd number, which would be a subset of the allowed durations (https://play.golang.org/p/K7-gKJ6yUh8)

---

yeah, i guess the first case is really just a no-op. i didn't realize about the second case, that's a good point. thanks!